### PR TITLE
Restructure history package

### PR DESCRIPTION
**Pull request by bernardjkim**

### CODE REVIEW

**Comment by bernardjkim** [comment]
Ignoring test files and godocs for now. Wanted to get some feedback first on if this would be a better approach than we had before.

- Events are no longer all the same struct type with a type field, now each event is defined by their own struct type and implement a Event interface. This way might be a bit more verbose but will be a bit more flexible if we need to add additional events in the future and reduces the number of fields that are needed for each event.

- Instead of defining another datatype for Cluster/Node/Probe/Events, just added a 'wrapper' datatype that embeds the generated protobuf message datatypes.

File: lib/history/event.go [hunk 1]
```diff
@@ -22,161 +22,208 @@ import (
...
-func newEvent(timestamp time.Time, eventType EventType) *Event {
-	return &Event{
-		timestamp: timestamp,
-		eventType: eventType,
-		metadata:  make(map[string]string),
+// Event represents a single timeline event.
+type Event interface {
+	// ToProto returns the event as a protobuf TimelineEvent message.
+	ToProto() *pb.TimelineEvent
+	// ToArgs returns the event as a list of arguments.
```
[the remark this answers is not in the document]
Comment (bernardjkim) [reply lib/history/event.go]: The `ToArgs` method is going to be used when inserting an event into the SQL database. Probably better to rename to something like `PrepareInsert`. Using an `Event` interface will allow each type of `Event` to have their own `ToArgs` or `PrepareInsert` methods depending on their required fields.

[the remark this answers is not in the document]
Comment (a-palchikov) [reply lib/history/event.go]: Hmm, I think if you inverted the direction, you could do away without the `interface{}`:
```go
type DataInserter interface {
	Insert(execer Execer) error
}
type Execer interface {
	Exec(query string, args ...interface{}) (sql.Result, error)
}
```
and implement it for events like:
```go
func (r *ClusterRecovered) Insert(execer Execer) error {
	const q = "insert into event(...) values(...)"
	return trace.Wrap(execer.Exec(q, r.Field1, r.Field2, ...))
}
```

[the remark this answers is not in the document]
Comment (bernardjkim) [reply lib/history/event.go]: Ah ok, I think I just over complicated things. This make a bit more sense.

**Pull request merged**